enhancements/aiops/20230120-Identify-prometheus-metrics-that-match-known-anomaly-chart-patterns.md

```diff
@@ -0,0 +1,57 @@
+# Title: 
+Identify prometheus metrics that match known anomaly chart patterns
+
+## Summary: 
+This enhancement proposal offers a new feature to match prometheus metrics to a set of known pre-defined anomaly chart patterns. This feature will make operators' life easier in the process of outage diagnosis.
+
+## Use case: 
+* Monitoring metrics in a multi-cluster setup is challenging. Operators could be responsible for hundreds of clusters and each cluster has tons of different metrics. When an outage occurs, it's difficult to identify root-cause metrics to assist outage diagnosis. Users need tools to help them reduce the scope and identify anomaly metrics faster.
+
+## Benefits: 
+* makes operators' life easier in the process of outage diagnosis.
+* improve user experience. Metric data will get condensed and categorized into insights that are much easier to digest.
+* reduce mean time to resolve outages. This method will shorten the time operators taken to find out root cause.
+* reduce economic loss.
+
+## Impact: 
+* No direct impact to existing system in terms of performance. However, if this new feature is deployed in a `proactively` in the future, then it might require some resources as it would have to continuously compute for all metrics.
+
+## Implementation details: 
+* pattern classification process. Inspired by this paper, https://netman.aiops.org/wp-content/uploads/2021/10/wch_ISSRE-1.pdf. 
+	a. A simple metric anomaly detection algorithm, that predicts every metric either normal or anomaly. An example method is Kolmogorov-Smirnov test.
+	b. A machine-learning or deep-learning model will be pre-trained for pattern classification. This would require some abnormal metric data collection and to apply data augmentation techniques. Once model is trained, it will be applied to anomaly metrics from step a and categorize them into a few pre-defined patterns.
```

Comment:
What is the plan to fetch abnormal metric data collection? Can you describe where this data will come from?

Reply:
real k8s metrics data with human labels